Online/Entity/OnlinePhysicalObject.cs

```diff
@@ -195,86 +189,98 @@ protected override void JoinImpl(OnlineResource inResource, EntityState initialS
-                                    RainMeadow.Debug("found in shortcuts? " + (ac2.realizedCreature != null && apo.world.game.shortcuts.betweenRoomsWaitingLobby.Any(v => v.creature.abstractCreature.GetAllConnectedObjects().Any(o => o.realizedObject == ac2.realizedCreature))));
+                                    RainMeadow.Debug($"early entity"); // room loading will place it
                                 }
                             }
-                            else
+                            else // nodedefined
                             {
-                                RainMeadow.Debug($"regular item, spawning off-room");
-                                apo.Realize(); 
-                                // and lets leave it at that, some creechur will connect to it and drag it in-room
+                                RainMeadow.Debug("node defined");
+                                apo.Move(topos);
+                                if (apo.realizedObject is Creature c)
+                                {
+                                    c.RemoveFromShortcuts();
+                                }
+                                if (apo is AbstractCreature ac2) // Creature.ChangeRoom didn't run, so we do it manually
+                                {
+                                    RainMeadow.Debug("creature moved");
+                                    if (ac2.realizedCreature == null || !ac2.realizedCreature.inShortcut)
+                                    {
+                                        RainMeadow.Debug($"spawning in shortcuts");
+                                        ac2.Realize();
+                                        ac2.realizedCreature.inShortcut = true;
+                                        ac2.world.game.shortcuts.CreatureEnterFromAbstractRoom(ac2.realizedCreature, ac2.world.GetAbstractRoom(topos), topos.abstractNode);
+                                    }
+                                    else
+                                    {
+                                        RainMeadow.Debug($"supposedly already spawning in shortcuts");
+                                        RainMeadow.Debug("found in shortcuts? " + (ac2.realizedCreature != null && apo.world.game.shortcuts.betweenRoomsWaitingLobby.Any(v => v.creature.abstractCreature.GetAllConnectedObjects().Any(o => o.realizedObject == ac2.realizedCreature))));
+                                    }
+                                }
+                                else
+                                {
+                                    RainMeadow.Debug($"regular item, spawning off-room");
+                                    apo.Realize();
+                                    // and lets leave it at that, some creechur will connect to it and drag it in-room
+                                }
                             }
                         }
```

Comment:
basically the whole thing got surrounded by an `if`

the diff is atrocious but that's it, no other changes